diff -Nuard gossip-0.6.orig/gossip.desktop.in gossip-0.6/gossip.desktop.in
--- gossip-0.6.orig/gossip.desktop.in	2003-11-03 20:18:57.000000000 +0100
+++ gossip-0.6/gossip.desktop.in	2003-11-23 02:23:43.791131080 +0100
@@ -7,6 +7,7 @@
 StartupNotify=true
 Terminal=false
 Type=Application
+Icon=gossip.png
 Categories=Application;Network;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gossip
