--- ./libloudermouth/lm-bs-transfer.c.org	2008-05-23 09:41:49.000000000 +0200
+++ ./libloudermouth/lm-bs-transfer.c	2012-11-30 23:21:47.221456706 +0100
@@ -428,7 +428,7 @@ bs_transfer_error (LmBsTransfer *transfe
 
 	error = g_error_new (lm_error_quark (),
 			     LM_BS_TRANSFER_ERROR_UNABLE_TO_CONNECT,
-			     error_msg);
+			     "%s", error_msg);
 	lm_bs_transfer_error (transfer, error);
 	g_error_free (error);
 }
--- ./src/gossip-account-widget-jabber.c.org	2008-05-08 09:48:16.000000000 +0200
+++ ./src/gossip-account-widget-jabber.c	2012-11-30 23:43:47.342765734 +0100
@@ -762,21 +762,21 @@ account_widget_jabber_register_cb (Gossi
 					     GTK_DIALOG_MODAL,
 					     GTK_MESSAGE_INFO,
 					     GTK_BUTTONS_CLOSE,
-					     str);
+					     "%s", str);
 
 		str = _("You should now be able to connect to your new account.");
-		gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (md), str);
+		gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (md), "%s", str);
 	} else {
 		str = _("Failed to register your new account settings.");
 		md = gtk_message_dialog_new (GTK_WINDOW (toplevel),
 					     GTK_DIALOG_MODAL,
 					     GTK_MESSAGE_ERROR,
 					     GTK_BUTTONS_CLOSE,
-					     str);
+					     "%s", str);
 		
 		if (error && error->message) {
 			gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (md),
-								  error->message);
+								  "%s", error->message);
 		}
 	}
 
