diff -aurN gossip-0.7.1.orig/gossip.schemas.in gossip-0.7.1/gossip.schemas.in
--- gossip-0.7.1.orig/gossip.schemas.in	2004-01-22 13:16:49.000000000 +0100
+++ gossip-0.7.1/gossip.schemas.in	2004-01-23 00:04:59.000000000 +0100
@@ -149,6 +149,12 @@
       <type>list</type>
       <list_type>string</list_type>
       <default>[]</default>
+       <locale name="C">
+          <short>Preset status messages</short>
+          <long>
+            Preset status messages.
+          </long>
+       </locale>
     </schema>
 
   </schemalist>  
