--- src/gossip-dbus.c	1 Feb 2004 00:36:37 -0000	1.2
+++ src/gossip-dbus.c	18 Aug 2004 19:52:03 -0000
@@ -30,7 +30,7 @@
 					    gpointer        user_data);
 
 
-static const char     *object_path[] = { "org", "imendio", "Gossip", NULL };
+static const char     *object_path = "/org/imendio/Gossip";
 
 static DBusObjectPathVTable vtable = {
 	unregistered_func,
@@ -45,12 +45,18 @@
 	DBusConnection *bus;
 	DBusError       error;
 
-	bus = dbus_bus_get_with_g_main (DBUS_BUS_SESSION, NULL);
+	bus = dbus_bus_get(DBUS_BUS_SESSION, NULL);
 	if (!bus) {
 		return FALSE;
 	}
   
 	dbus_error_init (&error);
+	dbus_connection_setup_with_g_main(bus, NULL);
+	if (dbus_error_is_set (&error)) {
+		g_warning ("Failed to setup dbus connection");
+		dbus_error_free (&error);
+		return FALSE;
+  }
 
 	dbus_bus_acquire_service (bus, GOSSIP_SERVICE, 0, &error);
 	if (dbus_error_is_set (&error)) {
diff -u -r1.2 gossip-dbus.h
--- src/gossip-dbus.h	1 Feb 2004 00:36:37 -0000	1.2
+++ src/gossip-dbus.h	18 Aug 2004 19:52:03 -0000
@@ -21,6 +21,8 @@
 #ifndef __GOSSIP_DBUS_H__
 #define __GOSSIP_DBUS_H__
 
+#include <dbus/dbus.h>
+#include <dbus/dbus-glib-lowlevel.h>
 #include <dbus/dbus-glib.h>
 
 #define GOSSIP_SERVICE "org.imendio.gossip"
